django-stubs/db/models/query.pyi

```diff
@@ -155,13 +156,14 @@ class NamedValuesListIterable(ValuesListIterable): ...
 class FlatValuesListIterable(BaseIterable):
     def __iter__(self) -> Iterator[Any]: ...
 
-class ValuesQuerySet(_BaseQuerySet[_T], Collection[_Row], Sized):
+class _ValuesQuerySet(Generic[_T, _Row], Collection[_Row], Reversible[_Row], QuerySet[_T], Sized): # type: ignore
+    def __len__(self) -> int: ...
     def __contains__(self, x: object) -> bool: ...
     def __iter__(self) -> Iterator[_Row]: ...  # type: ignore
     @overload  # type: ignore
     def __getitem__(self, i: int) -> _Row: ...
     @overload
-    def __getitem__(self: _QS, s: slice) -> _QS: ...
+    def __getitem__(self: _QS, s: slice) -> _QS: ... # type: ignore
```

Comment:
It was necessary to silence some error about self type being required for non-static-method. I am not sure the cause, but perhaps it was related to the error about MRO.